Fixed typos and grammar in lib/Syntax/README.md

NevinBR opened the pull request:
<!-- What's in this pull request? -->
Minor spelling and grammar corrections in the lib/Syntax readme.

NevinBR committed:
Fixed typos and grammar in README.md

xwu commented on lib/Syntax/README.md:
```diff
@@ -175,7 +175,7 @@ StructDeclSyntaxBuilder Builder;
 // We previously parsed a struct keyword, let's tell the builder to use it.
 Builder.useStructKeyword(StructKeyword);
 
-// Hm, we didn't see an identifier, but saw a left brace. Let's keep going.
```
This is fine as written.

NevinBR replied:
The original uses a non-standard variant spelling.

harlanhaskins replied:
I can take this or leave it. 'Hm' is just as correct as 'Hmm'.

NevinBR replied:
Okay I’m outvoted, we’ll keep it as “Hm”.

xwu commented on lib/Syntax/README.md:
```diff
@@ -163,8 +163,8 @@ struct YourStruct {}
 reasonably formed Syntax node (i.e. with no raw `nullptr`s) using what you've
 provided to the builder so far. Anything that you haven't supplied is marked as
-*missing*. This is essentially what the parser does so, looking forward to
-future adoption, the builders are designed with the parser in mind, with the
```
The punctuation here is more correct as written than in the revised version.

NevinBR replied:
I strongly disagree. There needs to be a comma between “does” and “so”.

After that it is more stylistic, and in my view reads better as revised.

harlanhaskins replied:
How about: "This is essentially what the parser does, so—looking forward to future adoption—the builders are designed with the parser in mind, with the

NevinBR replied:
Good thinking, Harlan, I’ll make it use em-dashes.

xwu replied:
Commas are more correct; em dashes are used to offset [shifts in thought](http://theweek.com/articles/460264/youre-using-that-dash-wrong):

> It's often a bad idea to use it where you would use a comma — unless the comma is at a clause boundary and you want to signify a greater break.

NevinBR replied:
Em dashes are used for [other things](https://en.wikipedia.org/wiki/Dash#Em_dash) as well, and would be perfectly correct here. However, I have reverted them to commas, and placed a semicolon between “does” and “so”, thus breaking up the run-on sentence after its first independent clause.

harlanhaskins commented:
Thanks for looking thoroughly thorough this document!

NevinBR committed:
Fixed typos and grammar in lib/Syntax/README.md

Incorporated suggestions from others.

NevinBR committed:
Fixed typos and grammar in lib/Syntax/README.md

Added a semicolon to a run-on sentence, and restored the original commas there.

CodaFi commented:
Will squash this down manually.

Thanks!

@swift-ci please smoke test

CodaFi commented:
⛵️
